### PR TITLE
[Refactor]Show catalogs enhancement and refactor show databases dql

### PR DESCRIPTION
**Pull request by adzfolc**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [x] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

1. Add component `from catalog_name` to dql `show databases` for dla connector
2. Refactor func handleShowCatalogs()
3. Add internal catalogs to `show catalogs;`

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 21 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/CatalogMgr.java|0|12|00.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowDbStmt.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/sql/ast/ShowCatalogsStmt.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|0|5|00.00%|

File: fe/fe-core/src/test/java/com/starrocks/analysis/ShowCatalogsStmtTest.java [hunk 1]
```diff
@@ -10,9 +10,9 @@
 public class ShowCatalogsStmtTest {
     @Test
     public void testShowCatalogsParserAndAnalyzer() throws AnalysisException {
-        final Analyzer analyzer = AccessTestUtil.fetchBlockAnalyzer();
-        ShowCatalogsStmt stmt = new ShowCatalogsStmt();
-        stmt.analyze(analyzer);
-        Assert.assertEquals("SHOW CATALOGS", stmt.toSql());
+        // final Analyzer analyzer = AccessTestUtil.fetchBlockAnalyzer();
```
Comment (DorianZheng) [inline fe/fe-core/src/test/java/com/starrocks/analysis/ShowCatalogsStmtTest.java:13]: why commented out this?

Comment (adzfolc) [reply fe/fe-core/src/test/java/com/starrocks/analysis/ShowCatalogsStmtTest.java:13]: > why commented out this?

图片
As mentioned in [pr 5963](https://github.com/StarRocks/starrocks/pull/5963#pullrequestreview-970285279), toSql() func is deleted in ShowCatalogsStmt.java , so UT in ShowCatalogsStmtTest is commented out.

Comment (DorianZheng) [reply fe/fe-core/src/test/java/com/starrocks/analysis/ShowCatalogsStmtTest.java:13]: ok, if we don't need it, just delete it

Comment (adzfolc) [reply fe/fe-core/src/test/java/com/starrocks/analysis/ShowCatalogsStmtTest.java:13]: > ok, if we don't need it, just delete it

ok

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 21 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/CatalogMgr.java|0|12|00.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowDbStmt.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/sql/ast/ShowCatalogsStmt.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|0|5|00.00%|

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java [hunk 2]
```diff
@@ -1556,15 +1554,11 @@ private void handleShowWorkGroup() {
...
-        List<List<String>> rows = Lists.newArrayList();
-        for (Map.Entry<String, Catalog> entry : catalogs.entrySet()) {
-            List<String> catalog = Lists.newArrayList();
-            catalog.add(entry.getKey());
-            Catalog value = entry.getValue();
-            catalog.add(((ExternalCatalog) value).getType());
-            catalog.add(value.getComment());
-            rows.add(catalog);
+        List<List<String>> rows = new ArrayList<>();
+        rows.add(Arrays.asList("default", "default", "internal catalog"));
```
Comment (caneGuy) [inline fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:1558]: how about moving this into CatalogMgr? cc @stephen-shelby

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/ShowCatalogsStmt.java [hunk 3]
```diff
@@ -22,25 +20,14 @@ public ShowCatalogsStmt() {
...
     }
 
     @Override
     public <R, C> R accept(AstVisitor<R, C> visitor, C context) {
-        return visitor.visitShowCatalogsStmt(this, context);
+        return visitor.visitShowCatalogsStatement(this, context);
     }
 
-    @Override
-    public String toSql() {
```
Comment (caneGuy) [inline fe/fe-core/src/main/java/com/starrocks/sql/ast/ShowCatalogsStmt.java:41]: why remove this?

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 4]
```diff
@@ -239,6 +235,11 @@ showAnalyzeStatement
     : SHOW ANALYZE
     ;
 
+// ------------------------------------------- Other Statement ---------------------------------------------------------
+showDatabasesStatement
```
Comment (caneGuy) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:239]: this has not been used now, am i right?

**Comment by caneGuy** [comment]
how about change the title as

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/ShowCatalogsStmt.java (hunk 3 above)
Comment (adzfolc) [reply fe/fe-core/src/main/java/com/starrocks/sql/ast/ShowCatalogsStmt.java:41]: As mentioned in pr 5963, toSql() func is deleted in ShowCatalogsStmt.java , so UT in ShowCatalogsStmtTest is commented out.

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 (hunk 4 above)
Comment (adzfolc) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:239]: showDatabasesStatement has been used now, but 'from catalog_name' won't make effects.It's downward compatible.

**Comment by adzfolc** [comment]
> how about changing the title to "Show catalogs enhancement and refactor show databases dql" since we have details for this in the summary which will also show in commit message later

Done.

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/ShowCatalogsStmt.java (hunk 3 above)
Comment (adzfolc) [reply fe/fe-core/src/main/java/com/starrocks/sql/ast/ShowCatalogsStmt.java:41]: > why remove this?

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java (hunk 2 above)
Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:1558]: yes, plz ref ResourceMgr.ResourceProcNode

File: fe/fe-core/src/main/java/com/starrocks/server/CatalogMgr.java [hunk 5]
```diff
@@ -79,4 +78,22 @@ public void replayDropCatalog(DropCatalogLog log) {
         dropCatalog(catalogName);
     }
 
+    public List<List<String>> showCatalogs() {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/server/CatalogMgr.java:81]: ditto

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 9 / 31 (29.03%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/ShowDbStmt.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/sql/ast/ShowCatalogsStmt.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/catalog/ExternalCatalog.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/catalog/InternalCatalog.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|0|4|00.00%|
|:large_blue_circle:|com/starrocks/server/CatalogMgr.java|8|18|44.44%|
|:large_blue_circle:|com/starrocks/common/proc/ProcService.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 9 / 31 (29.03%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/ShowDbStmt.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/sql/ast/ShowCatalogsStmt.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/catalog/ExternalCatalog.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/catalog/InternalCatalog.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|0|4|00.00%|
|:large_blue_circle:|com/starrocks/server/CatalogMgr.java|8|18|44.44%|
|:large_blue_circle:|com/starrocks/common/proc/ProcService.java|1|1|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/server/CatalogMgr.java [hunk 6]
```diff
@@ -21,8 +27,16 @@ public class CatalogMgr {
...
+    public static final ImmutableList<String> CATALOG_PROC_NODE_TITLE_NAMES = new ImmutableList.Builder<String>()
+            .add("Catalog").add("Type").add("Comment")
+            .build();
+
+    private final CatalogProcNode procNode = new CatalogProcNode();
+
     public CatalogMgr(ConnectorMgr connectorMgr) {
         this.connectorMgr = connectorMgr;
+        InternalCatalog internalCatalog = new InternalCatalog();
+        catalogs.put(InternalCatalog.DEFAULT_INTERNAL_CATALOG_NAME, internalCatalog);
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/server/CatalogMgr.java:39]: why create Internal catalog instance?

File: fe/fe-core/src/test/java/com/starrocks/analysis/ShowCatalogsStmtTest.java [hunk 7]
```diff
@@ -2,17 +2,6 @@
 
 package com.starrocks.analysis;
 
-import com.starrocks.common.AnalysisException;
-import com.starrocks.sql.ast.ShowCatalogsStmt;
-import org.junit.Assert;
-import org.junit.Test;
-
 public class ShowCatalogsStmtTest {
```
Comment (stephen-shelby) [inline fe/fe-core/src/test/java/com/starrocks/analysis/ShowCatalogsStmtTest.java:5]: add ut

File: fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java [hunk 8]
```diff
@@ -32,4 +33,11 @@ public void write(DataOutput out) throws IOException {
 
     public void readFields(DataInput in) throws IOException {
     }
+
+    /**
+     * Fill BaseProcResult with different properties in child resources
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java:38]: delete it

File: fe/fe-core/src/main/java/com/starrocks/catalog/InternalCatalog.java [hunk 9]
```diff
@@ -12,4 +15,9 @@ public InternalCatalog() {
     public InternalCatalog(String comment) {
         super(DEFAULT_INTERNAL_CATALOG_NAME, comment);
     }
+
+    public void getProcNodeData(BaseProcResult result) {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/catalog/InternalCatalog.java:19]: just hard code it

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java [hunk 10]
```diff
@@ -1556,17 +1553,15 @@ private void handleShowWorkGroup() {
...
-        }
-        resultSet = new ShowResultSet(showCatalogsStmt.getMetaData(), rows);
+        List<List<String>> rowSet = GlobalStateMgr.getCurrentState().getCatalogMgr().getCatalogsInfo();
+        rowSet.sort(new Comparator<List<String>>() {
+            @Override
+            public int compare(List<String> o1, List<String> o2) {
+                return o1.get(0).compareTo(o2.get(0));
+            }
+        });
+        // Only success
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:1563]: delete it

File: fe/fe-core/src/test/java/com/starrocks/analysis/ShowCatalogsStmtTest.java (hunk 7 above)
Comment (stephen-shelby) [reply fe/fe-core/src/test/java/com/starrocks/analysis/ShowCatalogsStmtTest.java:5]: ref CatalogStmtTest

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 11]
```diff
@@ -286,6 +282,10 @@ classifier
 
 // ------------------------------------------- Other Statement ---------------------------------------------------------
 
+showDatabasesStatement
+    : SHOW DATABASES ((LIKE pattern=string) | (WHERE expression) | (FROM catalogName=identifierOrString))?
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:286]:
```suggestion
    : SHOW  DATABASES ((FROM | IN) db=qualifiedName)? ((LIKE pattern=string) | (WHERE expression))?
```

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/ShowCatalogsStmt.java [hunk 12]
```diff
@@ -2,11 +2,9 @@
 
 package com.starrocks.sql.ast;
 
-import com.starrocks.analysis.Analyzer;
 import com.starrocks.analysis.ShowStmt;
 import com.starrocks.catalog.Column;
 import com.starrocks.catalog.ScalarType;
-import com.starrocks.common.AnalysisException;
 import com.starrocks.qe.ShowResultSetMetaData;
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/sql/ast/ShowCatalogsStmt.java:9]: Add it to StatementPlanner#supportedBynewPlaner and parser

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/ShowCatalogsStmt.java [hunk 13]
```diff
@@ -2,11 +2,9 @@
...
 
 public class ShowCatalogsStmt extends ShowStmt {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/sql/ast/ShowCatalogsStmt.java:10]: add it to StatementPlanner.supportedByNewPlanner

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 24 / 30 (80.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/ShowDbStmt.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/catalog/InternalCatalog.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/server/CatalogMgr.java|15|16|93.75%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/catalog/ExternalCatalog.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/common/proc/ProcService.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|5|5|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Analyzer.java [hunk 14]
```diff
@@ -183,7 +183,7 @@ public Void visitDropCatalogStatement(DropCatalogStmt statement, ConnectContext
         }
 
         @Override
-        public Void visitShowCatalogStatement(ShowCatalogsStmt statement, ConnectContext context) {
+        public Void visitShowCatalogsStatement(ShowCatalogsStmt statement, ConnectContext context) {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Analyzer.java:186]: delete it

**Comment by stephen-shelby** [comment]
fe will npe when fe restart using this patch , please fix it.

**Comment by adzfolc** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 25 / 29 (86.21%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/ShowDbStmt.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/server/CatalogMgr.java|16|17|94.12%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/catalog/ExternalCatalog.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/common/proc/ProcService.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|5|5|100.00%|

File: fe/fe-core/src/test/java/com/starrocks/analysis/ShowCatalogsStmtTest.java [hunk 15]
```diff
@@ -3,16 +3,59 @@
...
-        final Analyzer analyzer = AccessTestUtil.fetchBlockAnalyzer();
+    public void testShowCatalogsNormal() throws AnalysisException {
         ShowCatalogsStmt stmt = new ShowCatalogsStmt();
-        stmt.analyze(analyzer);
-        Assert.assertEquals("SHOW CATALOGS", stmt.toSql());
+        ShowExecutor executor = new ShowExecutor(ctx, stmt);
+        ShowResultSet resultSet = executor.execute();
+        ShowResultSetMetaData metaData = resultSet.getMetaData();
+        Assert.assertEquals(metaData.getColumn(0).getName(), "Catalog");
+        Assert.assertEquals(metaData.getColumn(1).getName(), "Type");
```
Comment (stephen-shelby) [inline fe/fe-core/src/test/java/com/starrocks/analysis/ShowCatalogsStmtTest.java:58]: nit: add some catalog then assert.

File: fe/fe-core/src/test/java/com/starrocks/analysis/ShowCatalogsStmtTest.java [hunk 16]
```diff
@@ -3,16 +3,59 @@
 package com.starrocks.analysis;
 
 import com.starrocks.common.AnalysisException;
+import com.starrocks.common.jmockit.Deencapsulation;
+import com.starrocks.qe.*;
```
Comment (stephen-shelby) [inline fe/fe-core/src/test/java/com/starrocks/analysis/ShowCatalogsStmtTest.java:7]: flat it

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 25 / 29 (86.21%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/ShowDbStmt.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/server/CatalogMgr.java|16|17|94.12%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/catalog/ExternalCatalog.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/common/proc/ProcService.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|5|5|100.00%|

File: fe/fe-core/src/test/java/com/starrocks/analysis/ShowCatalogsStmtTest.java (hunk 15 above)
Comment (adzfolc) [reply fe/fe-core/src/test/java/com/starrocks/analysis/ShowCatalogsStmtTest.java:62]: > nit: add some catalog then assert.

added

File: fe/fe-core/src/test/java/com/starrocks/analysis/ShowCatalogsStmtTest.java (hunk 16 above)
Comment (adzfolc) [reply fe/fe-core/src/test/java/com/starrocks/analysis/ShowCatalogsStmtTest.java]: > flat it

done

**Comment by stephen-shelby** [comment]
@HangyuanLiu PTAL , 3Q

**Review by stephen-shelby — APPROVED** [review]

**Review by HangyuanLiu — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/catalog/ExternalCatalog.java [hunk 17]
```diff
@@ -20,4 +22,8 @@ public ExternalCatalog(String name, String comment, Map<String, String> config)
     public String getType() {
         return type;
     }
+
+    public void getProcNodeData(BaseProcResult result) {
```
Comment (caneGuy) [inline fe/fe-core/src/main/java/com/starrocks/catalog/ExternalCatalog.java:26]: move this into Catalog？

**Review by caneGuy — APPROVED** [review]
other LGTM

**Pull request merged**